packages/browser/src/index.ts

```diff
@@ -25,4 +45,33 @@ export default class LogtoClient {
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
+
+  public async signIn(redirectUri: string) {
+    const { clientId, endpoint, requester, resources, scopes: originalScopes } = this.logtoConfig;
+    const discoveryEndpoint = new URL(discoveryPath, endpoint);
+    this.oidcConfig = await fetchOidcConfig(discoveryEndpoint.toString(), requester);
+
+    const { authorizationEndpoint } = this.oidcConfig;
+    const codeVerifier = generateCodeVerifier();
+    const codeChallenge = await generateCodeChallenge(codeVerifier);
+    const state = generateState();
+    const scopes = withReservedScopes(originalScopes).split(' ');
+
+    const signInUri = generateSignInUri({
+      authorizationEndpoint,
+      clientId,
+      redirectUri,
+      codeChallenge,
+      state,
+      scopes,
+      resources,
+    });
+
+    const logtoKey = getLogtoKey(clientId);
+    const logtoSignInSessionItem: LogtoSignInSessionItem = { redirectUri, codeVerifier, state };
+    const logtoValue = JSON.stringify(logtoSignInSessionItem);
+    sessionStorage.setItem(logtoKey, logtoValue);
```

Comment:
prefer to have a sessionStorage utiles to handle the set and get logic.  Wrap up all those, keyGenerate, JSONStringfy, JSON parse logic within the util method. simply call `logtoSesstion.set(),` `logtoSessiont.get()`

Reply:
extract two class methods to get and set the sign-in session item. PTAL.

See [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1644814026202089) on Slack.

Reply:
Do it in another PR next up.